### PR TITLE
feat: Introduce GitHub federated managed identities

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Add managed identities federated with GitHub to use a passwordless approach in pipeline definitions

### Motivation and context

Simplicity

### Type of changes

- [X] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [X] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [X] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: .identity/env/prod/terraform.tfvars [hunk 1]
```diff
@@ -1,8 +1,84 @@
...
+
+# # --- Managed Identities (new)
+
+ci_github_federations = [
+  {
+    repository = "io-infra"
+    subject    = "prod-ci"
+  },
+  {
+    repository = "io-sign"
```
Comment (gunzip) [inline .identity/env/prod/terraform.tfvars:66]: why there's io-sign here?

**Review by gunzip — COMMENTED** [review]
very good!

File: .identity/env/prod/terraform.tfvars (hunk 1 above)
Comment (Krusty93) [reply .identity/env/prod/terraform.tfvars:66]: We use a single identity for multiple repositories too keep things simple, so this list contains all the repos to federate

Comment (gunzip) [reply .identity/env/prod/terraform.tfvars:66]: I'd rather keep the io-sogn identity apart, ideally all io-sign infra should be moved into a separate subscription oneday. What do you think about it @lucacavallaro ?

**Review by gunzip — APPROVED** [review]

File: .identity/env/prod/terraform.tfvars (hunk 1 above)
Comment (Krusty93) [reply .identity/env/prod/terraform.tfvars:27]: @lucacavallaro After a chat with @gunzip, we agreed that moving that configuration to `io-sign` repository would be definitely better than this. However, it should be done when other parts of the infrastructure will be moved from `io-infra` to `io-sign` as well. I'm gonna add a comment to remind this

**Pull request merged**